src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,377 @@
+    List<SpeechRecognitionResult> results = response.getResultsList();
+
+    // if an output schema is available then use it or else use the schema for the given input record
+    Schema currentSchema;
+    if (outputSchema != null) {
+      currentSchema = outputSchema;
+    } else {
+      currentSchema = getSchema(input.getSchema(), config.getPartsField(), config.getTextField());
+    }
+
+    StructuredRecord.Builder outputBuilder = StructuredRecord.builder(currentSchema);
+
+    List<StructuredRecord> transcriptParts = new ArrayList<>();
+    StringBuilder completeTranscript = new StringBuilder();
+
+    Schema partsSchema = null;
+    if (config.getPartsField() != null && !config.getPartsField().isEmpty()) {
+      partsSchema = currentSchema.getField(config.getPartsField()).getSchema().getComponentSchema();
+    }
+
+    for (SpeechRecognitionResult result : results) {
+      if (config.getPartsField() != null && !config.getPartsField().isEmpty()) {
+        addTranscriptWithConfidence(partsSchema, transcriptParts, result.getAlternativesList());
+      }
+      if (config.getTextField() != null && !config.getTextField().isEmpty()) {
+        completeTranscript.append(result.getAlternativesList().get(0).getTranscript());
+      }
+    }
+
+    if (config.getPartsField() != null && !config.getPartsField().isEmpty()) {
+      outputBuilder.set(config.getPartsField(), transcriptParts);
+    }
+    if (config.getTextField() != null && !config.getTextField().isEmpty()) {
+      outputBuilder.set(config.getTextField(), completeTranscript.toString());
+    }
+    copyFields(input, outputBuilder);
+
+    emitter.emit(outputBuilder.build());
+  }
+
+  @Override
+  public void destroy() {
+    super.destroy();
+    try {
+      speech.close();
+    } catch (Exception e) {
+      // no-op
+    }
+  }
+
+  private Schema getSchema(Schema inputSchema, @Nullable String transcriptPartsField,
+                           @Nullable String transcriptTextField) {
+    List<Schema.Field> topLevel = new ArrayList<>();
+    boolean hasTranscriptField = false;
+    // one of the output field in which the transcript will be specified must be specified
+    if (transcriptPartsField != null && !transcriptPartsField.isEmpty()) {
+      topLevel.add(transcriptWithConfidenceSchema(transcriptPartsField));
+      hasTranscriptField = true;
+    }
+
+    if (transcriptTextField != null && !transcriptTextField.isEmpty()) {
+      topLevel.add(Schema.Field.of(transcriptTextField, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+      hasTranscriptField = true;
+    }
+
+    if (!hasTranscriptField) {
+      throw new IllegalArgumentException("Either 'Transcript Parts Field' or 'Transcript Text Field' or both must be " +
+                                           "specified.");
+    }
+
+    if (inputSchema != null && inputSchema.getFields() != null) {
+      topLevel.addAll(inputSchema.getFields());
```

Comment:
nit: add the existing fields first, with the new fields at the end. This is generally what we've done for new fields.